### PR TITLE
File with code to analyze added #1

### PR DESCRIPTION
**Pull request by Legosta**
just trying to do the homework

### CODE REVIEW

File: Untitled.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+
+
+
```
Comment (Legosta) [inline Untitled.py]: too many blank lines (3) before the function

File: Untitled.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        self.a=a
+    def __str__(self):
+        return self.a
+
+
+
+
+print("RESULT:")
+
+resultat = ''
```
Comment (Legosta) [inline Untitled.py]: it would be better to use english words in script, cause it looks like typo

File: Untitled.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    sheetIndex = 0
+    for X in range(1, s.nrows):
+        learner = s.row_values(X)
+        print(learner)
+        average=sum(learner)
+        out.append([learner[1], average])
+
+    return out
+
+def sum(learner):
```
Comment (Legosta) [inline Untitled.py]: built-in name "sum" as name of function

File: Untitled.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+debug = 1
+
+for s in a.sheets():
+    if (s.name == 'Python') == True:
+        out = sheet_data(s)
+        if debug == 1:
+            print(out)
+
+
+class PRESENT_RESULT  :
```
Comment (Legosta) [inline Untitled.py]: whitespace before ":"

File: Untitled.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
+
+
+
+def sheet_data(sheet):
+
+    out = list()
+
+    sheetIndex = 0
+    for X in range(1, s.nrows):
```
Comment (Legosta) [inline Untitled.py]: variable should be lowercase (PEP-8)

File: Untitled.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+for s in a.sheets():
+    if (s.name == 'Python') == True:
+        out = sheet_data(s)
+        if debug == 1:
+            print(out)
+
+
+class PRESENT_RESULT  :
+    def __init__(self, a):
+        self.a=a
```
Comment (Legosta) [inline Untitled.py]: missing whitespace around operator "="

File: Untitled.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+    sheetIndex = 0
+    for X in range(1, s.nrows):
+        learner = s.row_values(X)
+        print(learner)
+        average=sum(learner)
+        out.append([learner[1], average])
+
+    return out
+
```
Comment (Legosta) [inline Untitled.py]: expected 2 blank lines, found 1

File: Untitled.py (hunk 1 above)
Comment (Legosta) [inline Untitled.py]: too many blank lines (3)

File: Untitled.py [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+
+def sheet_data(sheet):
```
Comment (Legosta) [inline Untitled.py]: parameter "sheet" value is not used

File: Untitled.py [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+    out = list()
+
+    sheetIndex = 0
```
Comment (Legosta) [inline Untitled.py]: variables in function should be lowercase

File: Untitled.py (hunk 3 above)
Comment (Legosta) [inline Untitled.py]: built-in name "sum" is not permitted to use as name of function

File: Untitled.py [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+        average=sum(learner)
+        out.append([learner[1], average])
+
+    return out
+
+def sum(learner):
+    average = 0
+    for score in range(2, len(learner)) :
+        average = average+learner[score] // 2
+        if debug>0:
```
Comment (Legosta) [inline Untitled.py]: missing whitespace around operator ">"

File: Untitled.py (hunk 2 above)
Comment (Legosta) [inline Untitled.py]: it would be better to use english words in script, not russian words in latin

File: Untitled.py [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+            print(learner[score], average)
+    return average
+
+import xlrd
```
Comment (Legosta) [inline Untitled.py]: no modules named xlrd is found in script

File: Untitled.py (hunk 4 above)
Comment (Legosta) [inline Untitled.py]: whitespace before ":"

Comment (Legosta) [reply Untitled.py]: Class names should use CamelCase convention

File: Untitled.py [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+        return self.a
+
+
+
+
+print("RESULT:")
+
+resultat = ''
+
+for _ in range(1, len(out)):
```
Comment (Legosta) [inline Untitled.py]: name "out" cannot be defined

File: Untitled.py [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+class PRESENT_RESULT  :
+    def __init__(self, a):
+        self.a=a
+    def __str__(self):
+        return self.a
+
+
+
+
+print("RESULT:")
```
Comment (Legosta) [inline Untitled.py]: too many blank lines(4)

File: Untitled.py (hunk 6 above)
Comment (Legosta) [inline Untitled.py]: missing whitespace around operator "="

File: Untitled.py [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+    if (s.name == 'Python') == True:
+        out = sheet_data(s)
+        if debug == 1:
+            print(out)
+
+
+class PRESENT_RESULT  :
+    def __init__(self, a):
+        self.a=a
+    def __str__(self):
```
Comment (Legosta) [inline Untitled.py]: expected 1 blank line, found 0

File: Untitled.py [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+
+for s in a.sheets():
+    if (s.name == 'Python') == True:
+        out = sheet_data(s)
+        if debug == 1:
+            print(out)
+
+
+class PRESENT_RESULT  :
+    def __init__(self, a):
```
Comment (Legosta) [inline Untitled.py]: shadows name "a" from outer scope

File: Untitled.py (hunk 11 above)
Comment (Legosta) [reply Untitled.py]: expected 2 blank lines after class or function, found 1

File: Untitled.py [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+        learner = s.row_values(X)
+        print(learner)
+        average=sum(learner)
+        out.append([learner[1], average])
+
+    return out
+
+def sum(learner):
+    average = 0
+    for score in range(2, len(learner)) :
```
Comment (Legosta) [inline Untitled.py]: whitespace before ":"

File: Untitled.py [hunk 17]
```diff
@@ -0,0 +1,53 @@
+
+
+
+def sheet_data(sheet):
+
+    out = list()
```
Comment (Legosta) [inline Untitled.py]: shadows name "out" from outer scope

File: Untitled.py [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+
+
+
+
+print("RESULT:")
+
+resultat = ''
+
+for _ in range(1, len(out)):
+	resultat += str(out[_])
```
Comment (Legosta) [inline Untitled.py]: inconsistent indentation:  previous line used spaces, this line uses tabs

File: Untitled.py [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+
+def sheet_data(sheet):
+
+    out = list()
+
+    sheetIndex = 0
+    for X in range(1, s.nrows):
+        learner = s.row_values(X)
+        print(learner)
+        average=sum(learner)
```
Comment (Legosta) [inline Untitled.py]: missing whitespace around operator "="

File: Untitled.py (hunk 11 above)
Comment (Legosta) [reply Untitled.py]: module level import not at top of file

File: Untitled.py [hunk 20]
```diff
@@ -0,0 +1,53 @@
...
+            print(learner[score], average)
+    return average
+
+import xlrd
+
+a = xlrd.open_workbook("БИОЛОГИ_ таблица успеваемости осень 2018.xlsx")
+debug = 1
+
+for s in a.sheets():
+    if (s.name == 'Python') == True:
```
Comment (Legosta) [inline Untitled.py]: comparison to True should be 'if cond is True:' or 'if cond:'

File: Untitled.py (hunk 5 above)
Comment (Legosta) [inline Untitled.py]: it would be better if variable name is meaningful

File: Untitled.py (hunk 4 above)
Comment (Legosta) [reply Untitled.py]: this class is useless

File: Untitled.py (hunk 3 above)
Comment (agbragin) [reply Untitled.py]: Is **is** permitted, since the code above works without errors.

File: Untitled.py (hunk 12 above)
Comment (agbragin) [reply Untitled.py]: Is it actually not defined here?

File: Untitled.py (hunk 15 above)
Comment (agbragin) [reply Untitled.py]: Is it a problem here?

File: Untitled.py (hunk 17 above)
Comment (agbragin) [reply Untitled.py]: Is it a problem here?

**Pull request merged**